p2p/kademlia.py

```diff
@@ -451,9 +454,10 @@ def process(response):
 
         self.neighbours_callbacks[remote] = process
         try:
-            await asyncio.wait_for(event.wait(), k_request_timeout)
+            await wait_with_token(
```

Comment:
Since *every* `wait_with_token` has a `CancelToken` as one of it's arguments, it seems like all of these call sites could be cleaned up were you to be able to call this directly from the `token` instance, rather than as a standalone function that has to be imported in every module.

```python
async def wait_neighbours(self, remote, cancel_token):
    ...
    try:
        cancel_token.wait(event.wait(), timeout=k_request_timeout)
    ...
```

This probably involves moving the `CancelToken.wait` to something like `_wait()` unless there is a reason for that API to be public.

Reply:
I thought about that before, but I'm not sure I like the idea of having a method on CancelToken that waits for and returns the first on a list of futures -- it just doesn't feel like the right place to have such an API